code/modules/mob/living/living.dm

```diff
@@ -937,9 +953,191 @@ Sorry Giacom. Please don't be mad :(
 	else
 		src << "<span class='warning'>You don't have the dexterity to do this!</span>"
 	return
-
 /mob/living/proc/can_use_guns(var/obj/item/weapon/gun/G)
 	if (!IsAdvancedToolUser())
 		src << "<span class='warning'>You don't have the dexterity to do this!</span>"
 		return 0
-	return 1
\ No newline at end of file
+	return 1
+
+/mob/living/carbon/proc/update_stamina()
+	return
+
+/mob/living/carbon/human/update_stamina()

```

Comment:
Called when our staminaloss changed, to see we should be weakened and to update our health hud (since adjustStaminaloss() does not call updatehealth() or update_stat()